File: src/main/kotlin/codeacademy/bookingforum/app/report/ReportController.java
package codeacademy.bookingforum.app.report;

import codeacademy.bookingforum.app.configuration.ResponseObject;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.security.access.annotation.Secured;
import org.springframework.web.bind.annotation.*;
import org.springframework.web.context.request.WebRequest;

import javax.validation.Valid;
import java.util.List;

@RestController
@RequestMapping("/api/report")
public class ReportController {
    @Autowired
    ReportService reportService;

    @Secured({"ROLE_ADMIN","ROLE_USER","ROLE_SELLER"})
    @PostMapping("/new")
    @ResponseStatus(HttpStatus.CREATED)
    public ResponseObject report(@RequestBody @Valid ReportDto reportDto, WebRequest request) {
        return reportService.report(reportDto, request);
    }

    @Secured("ROLE_ADMIN")
    @GetMapping("/all")
    public List<ReportDto> getAll() {
        return reportService.getAll();
    }

    @Secured("ROLE_ADMIN")
    @DeleteMapping("/delete/{id}") // ID of report
    public ResponseObject delete(@PathVariable("id") Long id, WebRequest request) {
        return reportService.delete(id, request);
    }

}

File: src/main/kotlin/codeacademy/bookingforum/app/report/ReportController.kt
package codeacademy.bookingforum.app.report

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.PutMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/report")
class ReportController {

    @Autowired
    private lateinit var service: ReportService

    @GetMapping("/getAllReports")
    fun getAllReports(): List<ReportDto?> {
        return service.getAllReports()
    }

    @GetMapping("/getReport/{id}")
    fun getReportById(@PathVariable id: Long): ReportDto? {
        return service.getReportById(id)
    }

    @PostMapping("/postReport")
    fun postReport(@RequestBody report: ReportDto?) {
        service.postReport(report)
    }

    @PutMapping("/updateReport/{id}")
    fun updateReport(@PathVariable id: Long, @RequestBody report: ReportDto?): String {
        return service.updateReport(id, report)
    }

    @DeleteMapping("/deleteReport/{id}")
    fun deleteReport(@PathVariable id: Long): String {
        return service.deleteReport(id)
    }
}